projects/steering_digital_input/src/digital_input_config.c

```diff
@@ -0,0 +1,58 @@
+#include "digital_input_config.h"
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "can_transmit.h"
+#include "delay.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "misc.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "wait.h"
+
+// Set up all GPIO Addresses to each button
+// that recieves digital inputs
+static const GpioAddress s_steering_digital_input[NUM_STEERING_DIGITAL_INPUTS] = {
+  [STEERING_DIGITAL_INPUT_HORN] = { .port = GPIO_PORT_B, .pin = 0 },
+  [STEERING_DIGITAL_INPUT_RADIO_PPT] = { .port = GPIO_PORT_B, .pin = 1 },
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_FORWARD] = { .port = GPIO_PORT_B, .pin = 2 },
+  [STEERING_DIGITAL_INPUT_HIGH_BEAM_REAR] = { .port = GPIO_PORT_B, .pin = 3 },
+  [STEERING_DIGITAL_INPUT_REGEN_BRAKE_TOGGLE] = { .port = GPIO_PORT_B, .pin = 4 },
+  [STEERING_DIGITAL_INPUT_CC_TOGGLE] = { .port = GPIO_PORT_A, .pin = 0 },
+  [STEERING_DIGITAL_DIGITAL_INPUT_CC_INCREASE_SPEED] = { .port = GPIO_PORT_A, .pin = 1 },
+  [STEERING_DIGITAL_DIGITAL_INPUT_CC_DECREASE_SPEED] = { .port = GPIO_PORT_A, .pin = 2 },
+  [STEERING_DIGITAL_INPUT_CAN_TX] = { .port = GPIO_PORT_A, .pin = 2 },
+  [STEERING_DIGITAL_INPUT_CAN_RX] = { .port = GPIO_PORT_A, .pin = 2 },
+};
+
+void prv_callback_raise_event(const GpioAddress *address, void *context) {
+  GpioState state;
+  gpio_get_state(address, &state);
+  EventId *event = context;
+  // Raises an event and pushes it onto the queue
+  event_raise(*event, 0);
+}
+
+StatusCode steering_digital_input_init(SteeringDigitalInputConfiguration *storage) {

```

Comment:
the user of this module won't need to pass the storage in, if you put the lookup table here as well.